src/index.ts

```diff
@@ -370,6 +373,7 @@ class Spanner extends GrpcService {
     };
     this.directedReadOptions = directedReadOptions;
     this._observabilityOptions = options.observabilityOptions;
+    ensureInitialContextManagerSet();
```

Comment:
Can you please explain why do we need this here ?

Reply:
I  moved it to one place at initialization of the SpannerClient once and for all instead of always in startTrace. I found that it was spamming debug logs and also causing certain edge cases along with unnecessary latency. Doing it once and for all was anyways the main goal instead of every  call.